ci/go-publisher-ossf-slsa3.yml

```diff
@@ -0,0 +1,22 @@
+name: SLSA Go releaser
+on:
+  workflow_dispatch:
+  release:
+    types: [created]
+
+permissions: read-all
+
+jobs:
+  build:
+    permissions:
+      id-token: write # To sign.
+      contents: write # To upload release assets.
+      actions: read   # To read workflow path.
+    # If you need more configuration options, such as ldflag examples, 
+    # visit https://github.com/slsa-framework/slsa-github-generator#golang-projects.
+    uses: slsa-framework/slsa-github-generator/.github/workflows/builder_go_slsa3.yml@v1.0.0
```

Comment:
Please use full 40 character sha ref for hardened dependency

Reply:
Re-usable workflows are not be on-par with standard actions when using the OIDC token (https://docs.github.com/en/actions/deployment/security-hardening-your-deployments/about-security-hardening-with-openid-connect#understanding-the-oidc-token).

The workflow branch is not reported in the OIDC token for the re-usable workflow: it just has `wrkflow_path@ref`

This is a current limitation of the GH token, and we've made a feature request to address this with @josepalafox

Practically, this means that the hash is not guaranteed to be from the main branch, which means it's worse to pin by hash: any of the maintainers can push a different branch without code reviews, and users would be using it and it would pass the verification of the [slsa-framework/slsa-verifier](https://github.com/slsa-framework/slsa-verifier)

Until GH adds support, we've decided to pin by full version `v1.2.3` instead.

Hope you understand.

Reply:
This probably deserves a good comment noting why the hash isn't used in this case.

Reply:
Any suggestion? Trying to keep it simple for users, but not sure how much we need say.

Reply:
Ah, I see this gets included as a template in their repo? Then it makes sense to leave it off.

Reply:
Got it. Thanks for the explanation. I agree to leave it off in case of reusable workflows.